src/tactic/core.lean

```diff
@@ -1164,21 +1180,32 @@ For example,
 adds an instance `ring new_int`, defined to be the instance of `ring ℤ` found by `apply_instance`.
 
 Multiple instances can be added with `@[derive [ring, module ℝ]]`.
+
+This derive handler applies only to declarations made using `def`, and will fail on such a
+declaration if it is unable to derive an instance. It is run with higher priority than the built-in
+handlers, which will fail on `def`s.
 -/
-@[derive_handler] meta def delta_instance : derive_handler :=
-λ cls tp,
-(do tp' ← mk_const tp,
-   tgt ← to_expr ``(%%cls %%tp'),
-   (_, v) ← solve_aux tgt (delta_target [tp] >> apply_instance >> done),
-   v ← instantiate_mvars v,
-   nm ← get_unused_name $ tp ++
-     match tgt with
-     | expr.app (expr.const nm _) _ := nm
+@[derive_handler, priority 1000] meta def delta_instance : derive_handler :=
```

Comment:
Could you make this priority actually a little higher than the default 1000? Now it just has higher priority since it is declared later.
```suggestion
@[derive_handler, priority 2000] meta def delta_instance : derive_handler :=
```

Reply:
Oh, good catch, thanks. That was a testing fail.